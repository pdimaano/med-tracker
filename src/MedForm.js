import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  Button,
  Form,
  Label,
  Input,
} from "reactstrap";

/** Form for adding a medication.
 *
 *  Props:
 *  - addMed: call to add med in parent
 *
 *  State:
 *  - local state for each field on form
 *
 */

function MedForm({ addMed }) {
  const [form, setForm] = useState({
    day: "sunday",
    name: "",
    description: "",
  });
  const navigate = useNavigate();

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setForm((f) => ({
      ...f,
      [name]: value,
    }));
  };

  const handleSubmit = (evt) => {
    evt.preventDefault();
    let { day, ...data } = form;
    addMed(day, data);
    navigate("/");
  };

  const { day, name, description } = form;

  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="fw-bold text-center">Add Medication</CardTitle>
          <CardText>Add a Medication</CardText>

          <Form onSubmit={handleSubmit}>
            <div className="mb-3">
              <Label for="day">Day</Label>
              <Input
                type="select"
                name="day"
                id="day"
                value={day}
                onChange={handleChange}
              >
                <option>sunday</option>
                <option>monday</option>
                <option>tuesday</option>
                <option>wednesday</option>
                <option>thursday</option>
                <option>friday</option>
                <option>saturday</option>
              </Input>
            </div>

            <div className="mb-3">
              <Label for="name">Name</Label>
              <Input
                name="name"
                id="name"
                value={name}
                onChange={handleChange}
              />
            </div>

            <div className="mb-3">
              <Label for="description">Description</Label>
              <Input
                type="textarea"
                name="description"
                id="description"
                value={description}
                onChange={handleChange}
              />
            </div>

            <Button className="float-end btn btn-dark">Add Med</Button>
          </Form>
        </CardBody>
      </Card>
    </section>
  );
}

export default MedForm;
