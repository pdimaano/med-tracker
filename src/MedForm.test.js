import { screen, fireEvent, render, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import MedForm from "./MedForm";

it("renders without crashing", async function () {
  const { asFragment } = render(
    <MemoryRouter initialEntries={["/add"]}>
      <MedForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});