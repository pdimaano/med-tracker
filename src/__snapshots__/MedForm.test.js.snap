// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="fw-bold text-center card-title"
        >
          Add A Medication
        </div>
        <form
          class=""
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="day"
            >
              Day
            </label>
            <select
              class="form-select"
              id="day"
              name="day"
            >
              <option>
                Sunday
              </option>
              <option>
                Monday
              </option>
              <option>
                Tuesday
              </option>
              <option>
                Wednesday
              </option>
              <option>
                Thursday
              </option>
              <option>
                Friday
              </option>
              <option>
                Saturday
              </option>
            </select>
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="name"
            >
              Name
            </label>
            <input
              class="form-control"
              id="name"
              name="name"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="description"
            >
              Description
            </label>
            <textarea
              class="form-control"
              id="description"
              name="description"
            />
          </div>
          <button
            class="float-end btn btn-dark btn btn-secondary"
          >
            Add Med
          </button>
        </form>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
